homeassistant/components/nest/camera.py

```diff
@@ -73,6 +74,25 @@ def __init__(self, device: Device) -> None:
         self._stream_refresh_unsub: Callable[[], None] | None = None
         self._attr_is_streaming = CameraLiveStreamTrait.NAME in self._device.traits
         self.stream_options[CONF_EXTRA_PART_WAIT_TIME] = 3
+        self._rtsp_live_stream_trait = self._get_rtsp_live_stream_trait()
+
+    def _get_rtsp_live_stream_trait(self) -> CameraLiveStreamTrait | None:
+        """Check for stream prerequisites and return a CameraLiveStreamTrait or None."""
+        if not (self.supported_features & CameraEntityFeature.STREAM):
+            return None
+        if CameraLiveStreamTrait.NAME not in self._device.traits:
+            return None
```

Comment:
@allenporter These lines are not covered by the tests. From looking at the code, because of the way `self.supported_features` is defined, it looks like this test is actually equivalent to the test in lines 81-82 above. Can we just remove these two lines here?

Reply:
There seems to be some overlap between `self._attr_is_streaming`, the `supported_features` property, and the additional check here in `_get_rtsp_live_stream_trait()`. Can these be combined and set in `__init__`. Is there a reason why the supported_features property can't be set through `_attr_supported_features`?

Reply:
Setting `_attr_supported_features` up front sounds fine to me. This code was added before those attrs existed. Theoretically the device traits can change via pubsub messages, but if it did that would probably just be a bug anyway, so moving to _attr sounds better.

Reply:
Ok, I pushed a commit which just sets all of those things together in the `__init__`. Let me know if you'd prefer I submit the change to _attr as a separate PR.